src/include/btree.i

```diff
@@ -1052,8 +1052,10 @@ __wt_row_leaf_value(WT_PAGE *page, WT_ROW *rip, WT_ITEM *value)
      * See the comment in __wt_row_leaf_key_info for an explanation of the magic.
      */
     if ((v & 0x03) == WT_KV_FLAG) {
-        value->data = WT_PAGE_REF_OFFSET(page, WT_KV_DECODE_VALUE_OFFSET(v));
-        value->size = WT_KV_DECODE_VALUE_LEN(v);
+        if (value != NULL) {
```

Comment:
Similarly to before, either split this into two different functions or rename and update the comment for the function. The former is my preference.

Reply:
I had the same thought about splitting apart the function, and I futzed with it a bit, but nothing obvious came out of it. The function reasonably returns the ts/txn information, so there's considerable overlap, we'd almost need two copies of the function, one that returns ts/txn information and one that returns ts/txn information plus a value.

Reply:
We can create small helper functions and reuse those functions in the separated functions. I'll try that.

Reply:
Splitted to two functions